File: FinancesWebApp/src/app/shared/models/transaction.ts
export interface Transaction {
    id: number;
    transactionType: TransactionType;
    date: Date;
    title: string | null;
    labelId: number | null;
    price: number;
    rowVersion: string | null;
}

export interface TransactionWithLabel extends Transaction {
    labelName: string;
    labelColor: string,
}

export interface GroupedTransaction {
    date: Date;
    transactions: TransactionWithLabel[];
}

export enum TransactionType {
    Income = 0,
    Expense = 1
}

export interface TransactionView {
    priorBalance: number | null;
    oldestTransactionDate: Date | null;
    transactions: Transaction[];
}

export interface keyMetricData {
    total: number;
    bilanz: number;
    income: number;
    expense: number;
}

export interface AddOrEditTransaction {
    useCase: 'add' | 'edit';
    transactionId: number | null; // Is null if useCase = 'add'
    transactionType: TransactionType;
}

File: FinancesAPI/FinancesBackend/Transaction/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace FinancesBackend.Transaction.Models
{
    public sealed class Transaction
    {
        [Key] public int Id { get; set; }

        [Required] public Guid UserId { get; set; }

        [Required] public TransactionType TransactionType { get; set; }

        [Required] public DateTimeOffset Date { get; set; }

        public string? Title { get; set; }

        public int? LabelId { get; set; }

        [Required] public double Price { get; set; }

        [Timestamp] public byte[]? RowVersion { get; set; }
    }
}
